modules/vulnerabilities/unix/http/apache_spark_rce/manifests/install.pp

```diff
@@ -0,0 +1,64 @@
+# Class: apache_spark_rce::install
+# install process
+# https://archive.apache.org/dist/spark/spark-3.1.2/spark-3.1.2-bin-hadoop3.2.tgz
+# https://www.scala-lang.org/download/2.12.10.html
+class apache_spark_rce::install {
+  Exec { path => [ '/bin/', '/sbin/' , '/usr/bin/', '/usr/sbin/' ] }
+  $modulename = 'apache_spark_rce'
+
+  # Install required packages
+  # NOTE: once Debian updates insert scala 2.12+ into statement
+  ensure_packages(['openjdk-11-jdk'], { ensure => 'installed'})
+
+  $scaladeb = 'scala-2.12.10.deb'
+  $releasename = 'spark-3.1.2-bin-hadoop3.2.tgz'
+  $shortrelease = 'spark-3.1.2-bin-hadoop3.2'
+
+  $scalapart = ["${scaladeb}.partaa",
+    "${scaladeb}.partab",
+    "${scaladeb}.partac"]
+
+  $sparkpart = ["${releasename}.partaa",
+    "${releasename}.partab",
+    "${releasename}.partac",
+    "${releasename}.partad",
+    "${releasename}.partae"]
+
+  $pkgtobuild = [[$scalapart, $scaladeb], [$sparkpart, $releasename]]
+  $pkgtobuild.each |Array $pkg| {
+    apache_spark_rce::cpandbuild($pkg[0], $pkg[1])
+  }
+
+  $secgen_parameters = secgen_functions::get_parameters($::base64_inputs_file)
+  $user = $secgen_parameters['leaked_username'][0]
```

Comment:
in hindsight the parameter `leaked_username` is confusing -- sorry. This is meant to be usernames that are leaked to the attacker, not the Unix username to install the service under. I'll merge and sort this out.